lua/lean/infoview.lua

```diff
@@ -362,7 +367,7 @@ function Info:__set_diff_pin(params)
       use_widgets = options.use_widgets,
       parent = self
     }
-    self.__diff_renderer.__element = self.__diff_pin.__element
+    self.__diff_renderer.element = self.__diff_pin.__element
```

Comment:
Can we instead keep this private and fix the place looking for it?

Reply:
Do you mean to make `element` a private member of `BufRenderer`? (not sure I'm understanding correctly)

Reply:
So, I don't like doing `foo.bar = something` from outside of `foo`, I prefer immutable (or really persistent) objects which don't have their attributes changed over their lifetimes. We don't have that now, so we do need to do this in places, but if we do do this I'd like it to be to private attributes so it's even more clear it's something intended for change. Even better would be if we could create `__diff_renderer` here in the first place rather than having an existing one and mutating one of its attributes. Not sure if that's feasible or not or if there's a reason it needs to exist before this point in `Info`'s lifecycle.

I can look closer though if the above is more confusing than helpful (which it may be...).

Reply:
Just made it so that we use an element wrapper on top of the diff pin element, so we only need to modify its children (via a public function) similarly to how we currently do for the pins element. Let me know if it looks any better now.